lib/services/devices/deviceService.js

```diff
@@ -104,6 +106,75 @@ function createInitialEntityHandler(deviceData, newDevice, callback) {
     };
 }
 
+/**
+ * Creates the response handler for the initial entity creation request using NGSIv2.
+ * This handler basically deals with the errors that could have been rised during
+ * the communication with the Context Broker.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ * @param {Object} newDevice        Device object that will be stored in the database.
+ * @return {function}               Handler to pass to the request() function.
+ */
+function createInitialEntityHandlerNgsi2(deviceData, newDevice, callback) {
+    return function handleInitialEntityResponse(error, response, body) {
+        if (error) {
+            logger.error(context,
+                'ORION-001: Connection error creating inital entity in the Context Broker: %s', error);
+
+            alarms.raise(constants.ORION_ALARM, error);
+
+            callback(error);
+        } else if (response && response.statusCode === 201) {
+            alarms.release(constants.ORION_ALARM);
+            logger.debug(context, 'Initial entity created successfully.');
+            callback(null, newDevice);
+        } else {
+            var errorObj;
+
+            logger.error(context,
+                'Protocol error connecting to the Context Broker [%d]: %s', response.statusCode, body);
+
+            errorObj = new errors.EntityGenericError(deviceData.id, deviceData.type, body);
+
+            callback(errorObj);
+        }
+    };
+}
+
+/**
+ * Creates the response handler for the update entity request using NGSIv2. This handler basically deals with the errors
+ * that could have been rised during the communication with the Context Broker.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ * @param {Object} updatedDevice    Device object that will be stored in the database.
+ * @return {function}               Handler to pass to the request() function.
+ */
+function updateEntityHandlerNgsi2(deviceData, updatedDevice, callback) {
```

Comment:
Maybe existing updateEntityHandler() should be renamed to updateEntityHandlerNgsi1() ?

Reply:
updateEntityHandler() does not exist. As you can see in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/devices/deviceService.js#L562, for ngsiv1, the createInitialEntity function is called also to perform the update. I guess that the original developers took benefit of the fact that `Orion interprets APPEND as UPDATE if the entity already exists `.

Reply:
If updateEntityHandler() doesn't exist (I didn't check it... I though so applying a "paraellism" to the create case) then my original comment doesn't makes sense :)

NTC